derive/src/lib.rs

```diff
@@ -27,48 +28,167 @@ pub fn contract(input: proc_macro::TokenStream) -> TokenStream {
+                braced!(content in input);
+                let deployments = content
+                    .parse_terminated::<_, Token![,]>(Deployment::parse)?
+                    .into_iter()
+                    .collect();
+
+                Parameter::Deployments(deployments)
+            }
+            _ => {
+                return Err(ParseError::new(
+                    name.span(),
+                    format!("unexpected named parameter `{}`", name),
+                ))
+            }
+        };
+
+        Ok(param)
+    }
+}
+
+#[cfg_attr(test, derive(Debug, Eq, PartialEq))]
+struct Deployment {
+    network_id: u32,
+    address: Address,
+}
+
+impl Parse for Deployment {
+    fn parse(input: ParseStream) -> ParseResult<Self> {
+        let network_id = input.parse::<LitInt>()?.base10_parse()?;
+        input.parse::<Token![=>]>()?;
+        let address = {
+            let literal = input.parse::<LitStr>()?;
+            parse_address(&literal.value()).map_err(|err| ParseError::new(literal.span(), err))?
+        };
+
+        Ok(Deployment {
+            network_id,
+            address,
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    macro_rules! contract_args {
+        ($($arg:tt)*) => {{
+            use syn::parse::Parser;
+            <ContractArgs as Parse>::parse
+                .parse2(quote::quote! { $($arg)* })
+                .expect("failed to parse contract args")
+        }};
+    }
+
+    fn deployment(network_id: u32, address: &str) -> Deployment {
+        Deployment {
+            network_id,
+            address: parse_address(address).expect("failed to parse deployment address"),
+        }
+    }
+
+    #[test]
+    fn parse_contract_args() {
+        let args = contract_args!("path/to/artifact.json");
+        assert_eq!(args.artifact_path.value(), "path/to/artifact.json");
+    }
+
+    #[test]
+    fn parse_contract_args_with_parameter() {
+        let args = contract_args!("artifact.json", crate = foobar);
+        assert_eq!(args.parameters, &[Parameter::Crate("foobar".into())]);
+    }
+
+    #[test]
+    fn crate_parameter_accepts_keywords() {
+        let args = contract_args!("artifact.json", crate = crate);
+        assert_eq!(args.parameters, &[Parameter::Crate("crate".into())]);
+    }
+
+    #[test]
+    fn parse_contract_args_with_parameters() {
+        let args = contract_args!(
+            "artifact.json",
+            crate = foobar,
+            deployments {
+                1 => "0x000102030405060708090a0b0c0d0e0f10111213",
+                4 => "0x0123456789012345678901234567890123456789",
+            },
+        );
+        assert_eq!(
+            args.parameters,
+            &[
+                Parameter::Crate("foobar".into()),
+                Parameter::Deployments(vec![
+                    deployment(1, "0x000102030405060708090a0b0c0d0e0f10111213"),
+                    deployment(4, "0x0123456789012345678901234567890123456789"),
```

Comment:
Are you running a deterministic ganache?

Reply:
No, this is just a unit test to make sure that the proc macro parses the deployments correctly.